app/scripts/lib/account-tracker.js

```diff
@@ -586,25 +591,36 @@ export default class AccountTracker {
     try {
       const balances = await ethContract.balances(addresses, ethBalance);
 
+      const accounts = this.#getAccountsForChainId(chainId);
+      const newAccounts = {};
+      Object.keys(accounts).forEach((address) => {
+        if (!addresses.includes(address)) {
+          newAccounts[address] = { address, balance: null };
+        }
+      });
       addresses.forEach((address, index) => {
         const balance = balances[index] ? balances[index].toHexString() : '0x0';
         newAccounts[address] = { address, balance };
       });
 
       const { accountsByChainId } = this.store.getState();
-      accountsByChainId[chainId] = newAccounts;
-      this.store.updateState({ accountsByChainId });
-      if (chainId === this.getCurrentChainId()) {
-        this.store.updateState({ accounts: newAccounts });
-      }
+      this.store.updateState({
+        ...(chainId === this.getCurrentChainId() && {
+          accounts: newAccounts,
+        }),
+        accountsByChainId: {
+          ...accountsByChainId,
+          [chainId]: newAccounts,
+        },
+      });
     } catch (error) {
       log.warn(
         `MetaMask - Account Tracker single call balance fetch failed`,
         error,
       );
-      Promise.all(
+      Promise.allSettled(
```

Comment:
What are your thoughts on moving this to a separate PR? This seems unrelated to using `#` syntax for private methods, and might get lost here.

Reply:
Let's get this merged first then https://github.com/MetaMask/metamask-extension/pull/22306